src/imitation/algorithms/preference_comparisons.py

```diff
@@ -905,8 +927,30 @@ def __init__(
         self.optim = th.optim.AdamW(
             self._model.parameters(),
             lr=lr,
-            weight_decay=weight_decay,
         )
+        self.seed = seed
+        self.val_split = val_split
+
+        # TODO(juan) this could accept an arbitrary regularizer function
+        #  in the future if we wanted, with some changes in the __init__ arguments.
+        self.regularizer: Optional[regularization.Regularizer] = (
+            regularization.WeightDecayRegularizer(
+                initial_lambda=weight_decay,
+                optimizer=self.optim,
+                update_params_fn=weight_decay_updater
+                or regularization.ConstantParamScaler(),
+                logger=self.logger,
+            )
+            if weight_decay > 0
+            else None
+        )
+        if self.val_split == 0 and self.regularizer is not None:
```

Comment:
You could strengthen this to `(self.val_split == 0) == (self.regularizer is not None)` which will catch both this error and also the case where `val_split != 0` but regularizer is None. (Though maybe you want a different error message for that.)

Reply:
Smart use of XNOR, but maybe not so obvious to read for some people, and I agree that a different error message is in order

Reply:
Added a separate case for this